iOS/DuckDuckGo/DuckPlayer/NativeUI/Views/DuckPlayerMiniPillView.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  DuckPlayerMiniPillView.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import DesignResourcesKit
+
+/// A view that loads an image asynchronously with animation
+struct AnimatedAsyncImage: View {
+    let url: URL?
+    let width: CGFloat
+    let height: CGFloat
+
+    struct Constants {
+        static let backgroundColor: Color = .gray.opacity(0.3)
+    }
+
+    var body: some View {
+        if let url = url {
+            ZStack {
+                AsyncImage(url: url) { phase in
```

Comment:
Instead of using a if let (or guard) for the URL, wouldn't be possible to use the placeholder from AsyncImage?
From the [docs](https://developer.apple.com/documentation/swiftui/asyncimage): 
```
AsyncImage(url: URL(string: "https://example.com/icon.png")) { image in
    image.resizable()
} placeholder: {
    ProgressView()
}
.frame(width: 50, height: 50)
```